### PR TITLE
Use graphql GitHub API for getting initial list of PRs to process

### PR DESCRIPTION
**Pull request by akoeplinger**
This uses way less precious GitHub API calls. Right now this essentially just pre-filters
the PRs we want to process so we don't need to rewrite all of monors to GraphQL.

### CODE REVIEW

File: monors.py [hunk 1]
```diff
@@ -487,16 +486,71 @@ def main():
...
+                    collaborators(first: 100) {
+                        edges { node { login } }
+                    }
+                    pullRequests(last: 30, states: OPEN, orderBy: {field: UPDATED_AT, direction: ASC}) {
+                        edges {
+                            node {
+                                number
+                                mergeable
+                                title
+                                comments(last: 20) { edges { node { bodyText } } }
```
Comment (lewurm) [inline monors.py:509]: I believe 20 comments is a tight budget. I assume this graphql query counts as one API access, right?

File: monors.py [hunk 2]
```diff
@@ -487,16 +486,71 @@ def main():
...
+                    limit
+                    cost
+                    remaining
+                    resetAt
+                }
+                repository(owner: "mono", name: "mono") {
+                    collaborators(first: 100) {
+                        edges { node { login } }
+                    }
+                    pullRequests(last: 30, states: OPEN, orderBy: {field: UPDATED_AT, direction: ASC}) {
```
Comment (lewurm) [inline monors.py]: same here. we have almost 200 open PRs right now on mono/mono.

File: monors.py (hunk 1 above)
Comment (akoeplinger) [reply monors.py:509]: Sort of. GraphQL rate limiting is a bit more complex (read more here: https://developer.github.com/v4/guides/resource-limitations/#rate-limit). I checked this query and it returns a cost of `1` so it should be fine.

**Review by lewurm — APPROVED** [review]
looks good otherwise

File: monors.py (hunk 2 above)
Comment (akoeplinger) [reply monors.py]: We requested the last 50 PRs in the old version, I think I'll change to that. I think the reasoning was that the automerge command is most likely on a recent PR, not one from two years ago.

**Pull request merged**